### PR TITLE
docs(Storybook): document UsageGuidelines

### PR DESCRIPTION
**Pull request by talkor**
Used the new convention of storybook 7, let me know what you guys think.
Once we're ok with this let's continue documenting the rest of the components here

### CODE REVIEW

File: src/components/usage-guidelines/usage-guidelines.jsx
Comment (talkor) [inline src/components/usage-guidelines/usage-guidelines.jsx:1]: Personally I like PascalCase for components' filenames

File: src/components/usage-guidelines/usage-guidelines.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { BEMClass } from '../../helpers/utils/bem-helper';
-import './usage-guideline.scss';
+import './usage-guidelines.scss';
```
Comment (talkor) [inline src/components/usage-guidelines/usage-guidelines.jsx:4]: File name was incorrect (doesn't match the component's name)

File: .storybook/main.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import type { StorybookConfig } from '@storybook/react-webpack5';
 
 const config: StorybookConfig = {
-  stories: ['../src/**/*.stories.mdx', '../storybook/**/*.stories.mdx'],
+  stories: ['../src/**/*.mdx', '../src/**/*.stories.tsx', '../storybook/**/*.stories.mdx'],
```
Comment (talkor) [inline .storybook/main.ts:4]: Stories should now be in `*.stories.[tsx|js]` format and mdx files should not include `.stories`

File: src/components/usage-guidelines/README.md [hunk 3]
```diff
@@ -1,13 +0,0 @@
-# UsageGuideline
```
Comment (talkor) [inline src/components/usage-guidelines/README.md:1]: Added this documentation to the mdx file

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Proposed some changes here https://github.com/mondaycom/vibe-storybook-components/pull/24
Please, tell me what you think

**Review by SergeyRoyt — APPROVED** [review]

**Pull request merged**